resteasy-core/src/main/java/org/jboss/resteasy/core/ContextServletOutputStream.java

```diff
@@ -0,0 +1,175 @@
+package org.jboss.resteasy.core;
+
+import java.io.IOException;
+import java.io.OutputStream;
+
+import javax.servlet.ServletOutputStream;
+import javax.servlet.WriteListener;
+
+public class ContextServletOutputStream extends ServletOutputStream
+{
+   private ContextParameterInjector injector;
+   private ServletOutputStream delegate;
+
+   public ContextServletOutputStream(final ContextParameterInjector injector, final OutputStream delegate)
+   {
+      this.injector = injector;
+      this.delegate = (ServletOutputStream) delegate;
```

Comment:
Do we know for sure this will always be a `ServletOutputStream`? If so why do we accept an `OutputStream` and then just cast it?